### ISSUE TITLE
Proguard class name obfuscation not working properly when using Butterknife

### ISSUE THREAD

**Issue by markushi** [JakeWharton/butterknife#104]
Adding the Proguard configuration from http://jakewharton.github.io/butterknife/ to your project causes all class names to be kept during obfuscation.

Desired result: The class names should only be kept for classes using Butterknife.

Here is the Proguard configuration from your website:
```
-dontwarn butterknife.internal.**
-keep class **$$ViewInjector { *; }
-keepnames class * { @butterknife.InjectView *;}
```
The ``` -keepnames class * { @butterknife.InjectView *;} ``` seems to be the root cause of the problem.

Here is an alternative configuration which works well in my environment:
```
# Butterknife
-dontwarn butterknife.internal.**
-keep class butterknife.** { *; }
-keep class **$$ViewInjector { *; }

-keepclasseswithmembernames class * {
    @butterknife.InjectView <fields>;
}

-keepclasseswithmembernames class * {
    @butterknife.OnClick <methods>;
    @butterknife.OnEditorAction <methods>;
    @butterknife.OnItemClick <methods>;
    @butterknife.OnItemLongClick <methods>;
    @butterknife.OnLongClick <methods>;
}
```

**Comment by JakeWharton** [comment]
This will be fixed with #197 which will let you directly reference the generated class (like Dagger 2) and do full obfuscation with no config.

**Comment by ajinkyapote1** [comment]
I am using butterknife:7.0.1. I want my class names which are using butterknife to be obfuscated. what should be done?
reply ASAP.
thanks.
